react-example/src/components/EmbeddedForm.tsx

```diff
@@ -154,7 +156,8 @@ export const EmbeddedForm: FC<Props> = ({
     setTrackingEvent(true);
 
     const sessionData = {
-      userId: userId,
+      userId: emailRegex.test(userId) ? undefined : userId,
```

Comment:
you should be able to just pass in the value but determine which key to use dynamically. would also suggest making a helper/util method to do so.

Reply:
See https://www.freecodecamp.org/news/how-to-set-dynamic-object-properties-using-computed-property-names/

Reply:
Modified both the SDK and the sample app to incorporate support for 'Computed Property Names' wherever applicable.